utils/loader.py

```diff
@@ -0,0 +1,58 @@
+import os
+import pandas as pd
+from skimage import io
+from torch.utils.data import Dataset
+
+class HuaweiDataset(Dataset):
+    def __init__(self, root_dir=None, transform=None):
+        """
+        Args:
+            root_dir (string, optional): Directory with all the image folders, and 'Training_Info.csv'.
+            transform (callable, optional): Optional transform to be applied on a sample.
+        """
+        self.root_dir = root_dir
+        self.transform = transform
+        if root_dir is not None:
+            self.root_dir = root_dir
+        else:
+            # Use the default path, assumes repo was cloned alongside a `data` folder
+            self.root_dir = os.path.join(
+                os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                "data"
+            )
+            if not os.path.isdir(self.root_dir):
```

Comment:
I think this is too far indented?

Reply:
That was actually intentional, but you're right it makes far more sense outside the `else`